### PR TITLE
Generate a simple non-bootable firmware when not provided. (v2)

### PR DESCRIPTION
**Pull request by asiekierka**
Updated version of https://github.com/Arisotura/melonDS/pull/778 with community feedback.

### CODE REVIEW

File: src/Config.cpp [hunk 1]
```diff
@@ -59,6 +67,14 @@ ConfigEntry ConfigFile[] =
     {"DLDIEnable", 0, &DLDIEnable, 0, NULL, 0},
     {"DLDISDPath", 1, DLDISDPath, 0, "", 1023},
 
+    {"FirmwareUsername", 1, FirmwareUsername, 0, "MelonDS", 63},
```
Comment (RSDuck) [inline src/Config.cpp:70]: melonDS is the proper capitalisation

**Review by RSDuck — APPROVED** [review]

File: src/Config.cpp (hunk 1 above)
Comment (asiekierka) [reply src/Config.cpp:70]: Might be a good time to ask what the default username *should* be?

Comment (Arisotura) [reply src/Config.cpp:70]: if it's going to be melonDS it should atleast be capitalized correctly :P 

other than that, I'm currently out of ideas, other than stupid shit like 'Kagamine Len'

maybe we should run a poll somewhere?

File: src/SPI.cpp [hunk 2]
```diff
@@ -155,6 +159,52 @@ void Reset()
...
+    Firmware[UserSettings+0x50] = messageLength;
+}
+
+void Reset()
+{
+    if (Firmware) delete[] Firmware;
+    Firmware = NULL;
+
+    if (NDS::ConsoleType == 1)
+        strncpy(FirmwarePath, Config::DSiFirmwarePath, 1023);
```
Comment (DesperateProgrammer) [inline src/SPI.cpp:194]: Can we use sizeof()-1 as the limit?

Comment (asiekierka) [reply src/SPI.cpp]: This should be better.

File: src/SPI.cpp [hunk 3]
```diff
@@ -142,19 +146,75 @@ void Reset()
...
+    }
     else
     {
-        f = Platform::OpenLocalFile(firmbkp, "wb");
-        fwrite(Firmware, 1, FirmwareLength, f);
         fclose(f);
     }
+}
+
+void LoadUserSettingsFromConfig() {
```
Comment (WaluigiWare64) [inline src/SPI.cpp:174]: `{` on newline please.

```suggestion
void LoadUserSettingsFromConfig()
{
```

**Pull request merged**